tests/unit/jobserver/test_hash_utils.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from django.http import Http404
+
+from jobserver import hash_utils
+
+
+def test_roundtrip_1():
+    for m in range(2 ** 16):
+        assert hash_utils.unhash(hash_utils.hash(m)) == m
+
+
+def test_roundtrip_2():
+    for m_hash in range(2 ** 16):
+        h = hex(m_hash)[2:].rjust(4, "0")
+        assert hash_utils.hash(hash_utils.unhash(h)) == h
```

Comment:
Could you explain what this one is doing?  Particularly the definition of `h` is rather confusing to me!

Reply:
Have tried to explain notation in module docstring -- can you let me know whether it's clear enough?

Reply:
Not at all I'm afraid.

Things I would like explained in this test in particular:
* why are we running `hex()` on the int in `m_hash`?
* Why are we slicing the resulting string?
* Why are we padding that string out to 4 digits with zeros?
* And finally, what does the round trip through hash and unhash show us?

Reply:
The first three questions are answered by the output in the module docstring.  I tried to explain using words, but couldn't find a way to do so that wasn't more convoluted than showing an example.

Have added comments to the roundtrip tests which I hope clarifies things.